src/permutation.jl

```diff
@@ -0,0 +1,65 @@
+export PermutationTest
+
+function ptstats(x,y,f)
+    xy = vcat(x,y)
+    rx = 1 : length(x)
+    ry = length(xy)-length(y)+1 : length(xy)
+    T = typeof(f(x))  # is there a better way?  e.g., does a funtype() exist, like eltype()??
```

Comment:
What you're doing here is setting `T` to be the type of the result after applying `f` to `x`. So if `f(x)` returns 1, this will be `Int`, for example. Is this what you're after? I'm not sure what you mean regarding `eltype`.